pkg/controllers/controllers.go

```diff
@@ -106,7 +108,8 @@ func Register(ctx context.Context, systemNamespace, controllerName string, cfg c
 		klog.Infof("Starting helm controller in namespace %s", systemNamespace)
 	}
 
-	leader.RunOrDie(ctx, systemNamespace, "helm-controller-lock", appCtx.K8s, func(ctx context.Context) {
+	controllerLockName := controllerName + "-lock"
+	leader.RunOrDie(ctx, systemNamespace, controllerLockName, appCtx.K8s, func(ctx context.Context) {
```

Comment:
I suspect that if systemNamespace is blank here, the lock will actually go into the default namespace? But we should validate that, and perhaps allow configuring the lock namespace separate from the watched namespace?

Reply:
Yes, that is correct - if systemNamespace is blank the underlying `leader.RunOrDie` will try to use `kube-system` (which is the default in this case).

> perhaps allow configuring the lock namespace separate from the watched namespace?

I'm open to that, I can see some use-cases where that could be useful. Save that for a follow up effort in a new PR? I'd imagine that's introducing a new cli flag (and env var) to control that?

Reply:
yep!